src/common/telemetry_event/src/lib.rs

```diff
@@ -42,6 +43,27 @@ pub fn get_telemetry_risingwave_cloud_uuid() -> Option<String> {
     env::var(TELEMETRY_RISINGWAVE_CLOUD_UUID).ok()
 }
 
+static TELEMETRY_EVENT_REPORT_STASH: LazyLock<Mutex<Vec<PbEventMessage>>> =
+    LazyLock::new(|| Mutex::new(Vec::new()));
+
+pub async fn do_telemetry_event_report() {
+    const TELEMETRY_EVENT_REPORT_TYPE: &str = "event";
+    let url = (TELEMETRY_REPORT_URL.to_owned() + "/" + TELEMETRY_EVENT_REPORT_TYPE).to_owned();
+    let mut batch_message = PbBatchEventMessage { events: Vec::new() };
+
+    let mut stash_guard = TELEMETRY_EVENT_REPORT_STASH.lock().await;
+    for event in stash_guard.drain(..) {
+        batch_message.events.push(event);
+    }
+    drop(stash_guard);
+
+    post_telemetry_report_pb(&url, batch_message.encode_to_vec())
+        .await
+        .unwrap_or_else(|e| tracing::debug!("{}", e));
+}
+
+pub const TELEMETRY_EVENT_REPORT_INTERVAL: u64 = 10; // 10 seconds
```

Comment:
Is the size of the events accumulated in 10s controllable? If not, I am concerned that doing only time-based batching can cause memory pressure / OOM on the node. We might need to consider adding size-based batching as well.

Reply:
It makes sense. Let me fix.